pairing/pkg/cmd/pair.go

```diff
@@ -0,0 +1,228 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/codeready-toolchain/pairing/pkg/cmd/flags"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/spf13/cobra"
+)
+
+type PairingServiceInterface interface {
```

Comment:
If we replaced this interface with a simple `func` we could get rid of both `MockPairingService` and `PairingService` and just replace them with their respective `shouldPair` impls.

Additionally, if you could mock out the actual HTTP call in `shouldPair` func, we could also no longer require passing in the pairing `func` (or interface) to `Pair()` altogether and just use the actual impl also in the tests.